File: main/response/Response.cs
using main.payload;
using System.Text.Json.Serialization;

namespace main.response
{
    public class Response
    {
        [JsonPropertyName("status")]
        public ResponseStatus Status { get; set; }
        [JsonPropertyName("payload")]
        public Payload Payload { get; set; }
    }
}

File: java-client/src/main/java/cs/ubb/mpp/client/gateway/response/Response.java
package cs.ubb.mpp.client.gateway.response;

import cs.ubb.mpp.client.gateway.payload.Payload;

public class Response {
    private ResponseStatus status;
    private Payload payload;

    public Response(ResponseStatus status, Payload payload) {
        this.status = status;
        this.payload = payload;
    }

    public ResponseStatus getStatus() {
        return status;
    }

    public Payload getPayload() {
        return payload;
    }
}
